datadog_checks_base/datadog_checks/base/checks/openmetrics/mixins.py

```diff
@@ -146,6 +146,15 @@ def create_scraper_configuration(self, instance=None):
         # very high cardinality. Metrics included in this list will be silently
         # skipped without a 'Unable to handle metric' debug line in the logs
         config['ignore_metrics'] = instance.get('ignore_metrics', default_instance.get('ignore_metrics', []))
+        config['_ignored_metrics'] = {}
+        config['_ignored_patterns'] = {}
+
+        # Separate ignored metric names and ignored patterns in different maps for faster lookup later
+        for metric in config['ignore_metrics']:
+            if '*' in metric:
+                config['_ignored_patterns'][metric] = True
+            else:
+                config['_ignored_metrics'][metric] = True
```

Comment:
Can’t we use `sets` rather than `dictionnary` ?
It seems that the only value that we use is `True` and we only check for element existence.

```suggestion
        config['_ignored_metrics'] = set()
        config['_ignored_patterns'] = set()

        # Separate ignored metric names and ignored patterns in different maps for faster lookup later
        for metric in config['ignore_metrics']:
            if '*' in metric:
                config['_ignored_patterns'].add(metric)
            else:
                config['_ignored_metrics'].add(metric)
```

Reply:
indeed thanks!